### ISSUE TITLE
Disk space used by deleted objects is not being freed

### ISSUE THREAD

**Issue by sphuber** [leo-project/leofs#516]
This may be a complete noob question, being an absolute novice in using object stores, and raising a Github ticket may be the wrong approach, but I can not find the answer to my question anywhere. Apologies in advance.

I have setup Leofs 1.3.0 on an Ubuntu 16.04 virtual machine. Everything seems to be running fine with single storage node and I have been able to insert and delete objects into various buckets. However, after deleting the buckets and their objects, they are correctly no longer shown by the leofs-adm client, however the disk space is still being used. Even when the disk has completely ran out of space, Leofs won't clean up the unused space.

Now this may have to do with the fact that I am not sure what the procedure would be to tell Leofs to clean the disk space. I couldn't find anything concrete on the web. I have tried to use compact on the storage node. However, when querying the status, the compaction seems to be stuck with the following notice:

          current status: idling
          last compaction start: ____-__-__ __:__:__
          total targets: 8
       # of pending targets: 8
       # of ongoing targets: 0
       # of out of targets : 0

This has been the case for days and does not change. How does one clean unused disk space properly? Is the problem that I only have a single storage node perhaps?

**Comment by yosukehara** [comment]
@sphuber Thanks for your report. Let me know your operation history to reproduce this situation on my environment.

**Comment by sphuber** [comment]
@yosukehara Thank you for the quick reply. I just configured a vm from scratch to log the operation history while reproducing the issue, but now it seems to be working. I am not sure what I did differently the first time that caused compaction not to work. I will see if I can reproduce the behavior I encountered the first time. Could it be because I did not register the buckets I was using to store objects in, in the /etc/hosts file? Or maybe, that I deleted the bucket itself, before running the compaction. Anyway, I will test these two hypotheses and report back.

**Comment by sphuber** [comment]
I have tried multiple things to reproduce the original situation but to no avail. In the environment that produced the error, I used a combination of the administration tool, s3cmd and the python sdk 'boto' to insert and delete objects into various buckets. The combination of some actions may have been the culprit but I cannot remember exactly what I did. This ticket can be closed for now and I will reopen it when I encounter the issue again
